geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/io/fs/HadoopDelegate.scala

```diff
@@ -0,0 +1,74 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.utils.io.fs
+
+import java.io.InputStream
+
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path, PathFilter}
+import org.locationtech.geomesa.utils.io.fs.FileSystemDelegate.FileHandle
+import org.locationtech.geomesa.utils.io.fs.HadoopDelegate.HadoopFileHandle
+
+/**
+  * Delegate allows us to avoid a runtime dependency on hadoop
+  */
+class HadoopDelegate extends FileSystemDelegate {
+
+  import HadoopDelegate.HiddenFileFilter
+  import org.apache.hadoop.fs.{FileStatus, LocatedFileStatus, Path}
+
+  private val conf = new org.apache.hadoop.conf.Configuration()
+  // use the same property as FileInputFormat
+  private val recursive = conf.getBoolean("mapreduce.input.fileinputformat.input.dir.recursive", false)
+
+  // based on logic from hadoop FileInputFormat
+  override def interpretPath(path: String): Seq[FileHandle] = {
+    val p = new Path(path)
+    // TODO close filesystem?
```

Comment:
@jahhulbert-ccri thoughts?

Reply:
hmmm i haven't been doing this. I think maybe I should now that I look at it. Seems there is some caching and deleting going on perhaps. We should look into it. I don't think it would hurt. Maybe we can troll hadoop for some more examples of how they use it.

Reply:
i might have to refactor things here - we need to keep the file system around to open connections with, so this would have to be a CloseableSeq or something